dLux/apertures.py

```diff
@@ -0,0 +1,1430 @@
+        ----------
+        vertices : Matrix, meters
+            The vertices loaded from the WebbPSF module. 
+
+        Returns
+        -------
+        x, y, angles : tuple 
+            The vertices in normalised positions and wrapped so that 
+            they can be used in the generation of the compound aperture.
+            The `x` is the x coordinates of the vertices, the `y` is the 
+            the y coordinates and `angles` is the angle of the vertex. 
+        """
+        _x = (vertices[:, 0] - np.mean(vertices[:, 0]))
+        _y = (vertices[:, 1] - np.mean(vertices[:, 1]))
+
+        _angles = np.arctan2(_y, _x)
+        _angles += 2 * np.pi * (np.arctan2(_y, _x) < 0.)
+
+        # By default the `np.arctan2` function returns values within the 
+        # range `(-np.pi, np.pi)` but comparisons are easiest over the 
+        # range `(0, 2 * np.pi)`. This is where the logic implemented 
+        # above comes from. 
+
+        order = np.argsort(_angles)
+
+        x = self._wrap(_x, order)
+        y = self._wrap(_y, order)
+        angles = self._wrap(_angles, order).at[-1].add(2 * np.pi)
+
+        # The final `2 * np.pi` is designed to make sure that the wrap
+        # of the first angle is within the angular coordinate system 
+        # associated with the aperture. By convention this is the
+        # range `angle[0], angle[0] + 2 * np.pi` what this means in 
+        # practice is that the first vertex appearing in the array 
+        # is used to chose the coordinate system in angular units. 
+
+        return x, y, angles
+
+
+    def _offset(self : Layer, vertices : Matrix) -> tuple:
+        """
+        Get the offsets of the coordinate system.
+
+        Parameters
+        ----------
+        vertices : Matrix 
+            The unprocessed vertices loaded from the JWST data file.
+            The correct shape for this array is `vertices.shape == 
+            (2, number_of_vertices)`. 
+        pixel_scale : float, meters
+            The physical size of each pixel along one of its edges.
+
+        Returns 
+        -------
+        x_offset, y_offset : float, meters
+            The x and y offsets in physical units. 
+        """
+        x_offset = np.mean(vertices[:, 0])
+        y_offset = np.mean(vertices[:, 1])
+        return x_offset, y_offset
+
+
+    # TODO: number_of_pixels can be moved out as a parameter
+    def _rad_coordinates(self : Layer) -> tuple:
+        """
+        Generates the vectorised coordinate system associated with the 
+        aperture.
+
+        Parameters
+        ----------
+        phi_naught : float 
+            The angle substending the first vertex. 
+
+        Returns 
+        -------
+        rho, theta : tuple[Tensor]
+            The stacked coordinate systems that are typically passed to 
+            `_segments` to generate the segments.
+        """
+        cartesian = self._coordinates()
+        positions = cartesian_to_polar(cartesian)
```

Comment:
Should be replaced with `utils.coordinates.get_polar_coordinates()`

Reply:
The minus sign is the magic that makes this work :laughing:, I should be able to find a work around though.